### PR TITLE
Update knockout.contextmenu.js

### PR DESCRIPTION
**Pull request by jlaustill**
This update will make it work with requirejs and nodejs, as well as standalone.

### CODE REVIEW

**Comment by nescalante** [comment]
identation was broken, can you fix that and move it back again to 2 spaces as the main identation?

**Comment by jlaustill** [comment]
Absolutely!  I'm super impressed with your build setup, so I'm going to spend my afternoon going over it actually.  I wasn't expecting that at all :)

**Comment by nescalante** [comment]
thanks! 😃 let me know when you are ready and I will take a look.

**Comment by jlaustill** [comment]
I'm stumped on this one.  I'm a die hard jslint user, but jshint is complaining about something that jslint doesn't even complain about here, any ideas?

**Comment by nescalante** [comment]
any chance to squash commits when finished?

**Comment by jlaustill** [comment]
Of course :)

File: src/knockout.contextmenu.js [hunk 1]
```diff
@@ -1,15 +1,21 @@
...
+   Issues: https://github.com/nescalante/knockout.contextmenu/issues
+   License: MIT */
+/*globals define,module,exports*/
+(function (factory) {
+  'use strict';
+  if (define !== 'undefined' && typeof define === 'function' && define.amd) {
+    // AMD. Register as an anonymous module.
+    define(['knockout'], factory);
+  } else if (typeof module !== undefined + '' && module.exports) {
+    factory(module.exports || exports);  // module.exports is for Node.js
```
Comment (nescalante) [inline src/knockout.contextmenu.js]: shouldn't it be `module.exports || exports = factory`?

File: src/knockout.contextmenu.js [hunk 2]
```diff
@@ -1,15 +1,21 @@
...
-if (typeof ko !== undefined + '' && typeof document !== undefined + '') {
-  bindContextMenu(ko, document);
-}
-
-if (typeof module !== undefined + '' && module.exports) {
-  module.exports = bindContextMenu;
-}
-
-function bindContextMenu(ko, document) {
+/* knockout.contextmenu v0.4.4
```
Comment (nescalante) [inline src/knockout.contextmenu.js]: no need to put this comments here

File: src/knockout.contextmenu.js [hunk 3]
```diff
@@ -1,15 +1,21 @@
...
+  if (define !== 'undefined' && typeof define === 'function' && define.amd) {
+    // AMD. Register as an anonymous module.
+    define(['knockout'], factory);
+  } else if (typeof module !== undefined + '' && module.exports) {
+    factory(module.exports || exports);  // module.exports is for Node.js
+  } else {
+    // Browser globals
+    factory(ko);
+  }
+}(function (ko) {
```
Comment (nescalante) [inline src/knockout.contextmenu.js]: adding an `undefined` second property here will save a couple of bytes on the bundle

**Comment by jlaustill** [comment]
Ok, I figured out what's going on here, when I clone to windows, it changes all the line endings and seriously messes with the build process.  I'm sorry, I'll clone this to my mac and work from there.  I can just delete this pull request and start over if you'd like, this is a mess.

**Pull request closed without merging**